domain/consensus/consensus.go

```diff
@@ -183,11 +184,38 @@ func (s *consensus) GetPruningPointUTXOSet(expectedPruningPointHash *externalapi
 	return serializedUTXOSet, nil
 }
 
-func (s *consensus) SetPruningPointUTXOSet(serializedUTXOSet []byte) error {
+func (s *consensus) SetPruningPoint(newPruningPoint *externalapi.DomainBlock, serializedUTXOSet []byte) error {
 	s.lock.Lock()
 	defer s.lock.Unlock()
 
-	return s.consensusStateManager.SetPruningPointUTXOSet(serializedUTXOSet)
+	selectedTip, err := s.headersSelectedTipStore.HeadersSelectedTip(s.databaseContext)
+	if err != nil {
+		return err
+	}
+
+	expectedNewPruningPointHash, err := s.pruningManager.CalculateIndependentPruningPoint(selectedTip)
```

Comment:
Here we calculatePruningPoint according to selectedTip, while in another place we do according to virtual.
Wouldn't this create different pruning points whether you are IBDing or not?

Reply:
In UpdatePruningPointByVirtual it doesn't update pruning point unless `if virtual.BlueScore() <= currentPBlueScore+pm.finalityInterval {`